multiply defined xmg_dont_cares_optimization

When linking against commit 23401c120eb900e8c12b9fa14c182a7a519561bc of mockturtle, we get the following error:

```sh
...
../libfoo/libfoo.a(bar.cc.o): In function `mockturtle::xmg_dont_cares_optimization(mockturtle::xmg_network const&)':
bar.cc:(.text+0x5cb2): multiple definition of `mockturtle::xmg_dont_cares_optimization(mockturtle::xmg_network const&)'
CMakeFiles/foo.dir/main.cc.o:main.cc:(.text+0x5cb2): first defined here
../libfoo/libfoo.a(baz.cc.o): In function `mockturtle::xmg_dont_cares_optimization(mockturtle::xmg_network const&)':
baz.cc:(.text+0x5cb2): multiple definition of `mockturtle::xmg_dont_cares_optimization(mockturtle::xmg_network const&)'
CMakeFiles/foo.dir/main.cc.o:main.cc:(.text+0x5cb2): first defined here
../libfoo/libfoo.a(qux.cc.o): In function `mockturtle::xmg_dont_cares_optimization(mockturtle::xmg_network const&)':
qux.cc:(.text+0x5cb2): multiple definition of `mockturtle::xmg_dont_cares_optimization(mockturtle::xmg_network const&)'
CMakeFiles/foo.dir/main.cc.o:main.cc:(.text+0x5cb2): first defined here
collect2: error: ld returned 1 exit status
```

The root cause seems to be that `xmg_dont_cares_optimization` is a free-standing non-template non-static non-inline function in a header, so it ends up exported in multiple translation units. Is this intended to be `inline`?

--- a/include/mockturtle/algorithms/xmg_optimization.hpp
+++ b/include/mockturtle/algorithms/xmg_optimization.hpp
@@ -51,7 +51,7 @@ namespace mockturtle
  * If a MAJ gate is satisfiability don't care for assignments 000 and 111, it can be
  * replaced by an XNOR gate.
  */
-xmg_network xmg_dont_cares_optimization( xmg_network const& xmg )
+inline xmg_network xmg_dont_cares_optimization( xmg_network const& xmg )
 {
   node_map<xmg_network::signal, xmg_network> old_to_new( xmg );
 

